applications/IgaApplication/custom_utilities/brep_topology/brep_edge.cpp

```diff
@@ -0,0 +1,151 @@
+    {
+        auto spans = m_node_curve_geometry_3d->Spans();
+
+        Properties::Pointer this_property = rModelPart.pGetProperties(rPropertiesId);
+
+        for (int i = 0; i < spans.size(); ++i)
+        {
+            ANurbs::Interval<double> domain(spans[i].T0(), spans[i].T1());
+
+            int number_of_points = m_node_curve_geometry_3d->Degree() + 1;
+            auto integration_points = ANurbs::IntegrationPoints<double>::Points1(number_of_points, domain);
+
+            ANurbs::CurveShapeEvaluator<double> shape(m_node_curve_geometry_3d->Degree(), rShapeFunctionDerivativesOrder);
+
+            for (int j = 0; j < integration_points.size(); ++j)
+            {
+                Vector local_coordinates(1);
+                local_coordinates[0] = integration_points[j].t;
+
+                shape.Compute(m_node_curve_geometry_3d->Knots(), integration_points[j].t);
+
+                Vector N_0 = ZeroVector(shape.NbNonzeroPoles());
+                Matrix N_1 = ZeroMatrix(shape.NbNonzeroPoles(), 1);
+
+                Element::GeometryType::PointsArrayType non_zero_control_points;
+                //std::vector<Node<3>::Pointer> cps;
+                //std::vector<int> cp_ids;
+                for (int m = shape.FirstNonzeroPole(); m < shape.LastNonzeroPole() + 1; ++m)
+                {
+                    //cp_ids.push_back(m_node_curve_geometry_3d->Node(m)->Id());
+                    //cps.push_back(m_node_curve_geometry_3d->Node(m));
+                    non_zero_control_points.push_back(m_node_curve_geometry_3d->Node(m));
+
+                    N_0(m) = shape(0, m);
+
+                    N_1(m, 0) = shape(1, m);
+                }
+                if (rType == "element")
+                {
+                    int id = 0;
+                    if (rModelPart.GetRootModelPart().Elements().size()>0)
+                        id = rModelPart.GetRootModelPart().Elements().back().Id() + 1;
+                    
+                    auto element = rModelPart.CreateNewElement(rName, id, non_zero_control_points, this_property);
+
+                    element->SetValue(SHAPE_FUNCTION_VALUES, N_0);
+                    element->SetValue(SHAPE_FUNCTION_LOCAL_DERIVATIVES, N_1);
+                    element->SetValue(INTEGRATION_WEIGHT, integration_points[j].weight);
+                }
+                if (rType == "condition")
+                {
+                    int id = 0;
+                    if (rModelPart.GetRootModelPart().Conditions().size()>0)
+                        int id = rModelPart.GetRootModelPart().Conditions().back().Id() + 1;
+                    auto condition = rModelPart.CreateNewCondition(rName, id, non_zero_control_points, this_property);
+
+                    condition->SetValue(SHAPE_FUNCTION_VALUES, N_0);
+                    condition->SetValue(SHAPE_FUNCTION_LOCAL_DERIVATIVES, N_1);
+                    condition->SetValue(INTEGRATION_WEIGHT, integration_points[j].weight);
+                }
```

Comment:
@philbucher, do you see the possibility to somehow template the condition and element?